src/util/irep.h

```diff
@@ -104,6 +96,87 @@ const irept &get_nil_irep();
 /// * \ref irept::dt::sub : A vector of \ref irept which is used to store
 ///   ordered but unnamed children.
 ///
+/// * \ref ref_count : if sharing is activated, this is used to count the number
+///   of references to a node.
+///
+/// * \ref hash_code : if HASH_CODE is activated, this is used to cache the
+///   result of the hash function.
+template<typename treet>
+class tree_nodet
+{
+public:
+  // These are not stable.
+  typedef std::vector<treet> subt;
+
+  // named_subt has to provide stable references; with C++11 we could
+  // use std::forward_list or std::vector< unique_ptr<T> > to save
+  // memory and increase efficiency.
+
+#ifdef NAMED_SUB_IS_FORWARD_LIST
+  typedef std::forward_list<std::pair<irep_namet, irept>> named_subt;
+#else
+  typedef std::map<irep_namet, treet> named_subt;
+#endif
+
+  friend treet;
+
+#ifdef SHARING
+  unsigned ref_count = 1;
+#endif
+
+  /// This irep_idt is the only place to store data in an irep, other than
```

Comment:
"the only place to store data in an irep" -> "the only place to store data in a tree node"

Reply:
Remove ", other than the mere nesting structure", as it isn't relevant to `tree_nodet`. You could have a similar comment in `irept` somewhere if you want. (I added this comment a while ago because I found `irept` so hard to understand, so I approve of you making it clearer.)